cpp/src/strings/utilities.cuh

```diff
@@ -77,6 +78,34 @@ std::unique_ptr<column> make_offsets_child_column( InputIterator begin, InputIte
     return offsets_column;
 }
 
+/**
+ * @brief Utility to create a null mask for a strings column using a custom function.
+ *
+ * @tparam BoolFn Function should return true/false given index for a strings column.
+ * @param strings_count Number of strings for the column.
+ * @param bfn The custom function used for identifying null string entries.
+ * @param mr Memory resource to use.
+ * @param stream Stream to use for any kernel calls.
+ * @return Pair including null mask and null count
+ */
+template <typename BoolFn>
+std::pair<rmm::device_buffer,cudf::size_type> make_null_mask( cudf::size_type strings_count,
+    BoolFn bfn,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+    cudaStream_t stream = 0)
+{
+    auto valid_mask = valid_if( static_cast<const bit_mask_t*>(nullptr),
+                                bfn, strings_count, stream );
+    auto null_count = valid_mask.second;
+    rmm::device_buffer null_mask;
+    if( null_count > 0 )
+        null_mask = rmm::device_buffer(valid_mask.first,
```

Comment:
The latest valid_if already returns a device_buffer and a count. So why do you need `make_null_mask` at all?